File: frontend/src/types/job.ts
export interface Job {
  id: string;
  candidate_id: string;
  title: string;
  company: string;
  url: string | null;
  raw_description: string;
  seniority: string;
  location: string | null;
  salary_range: string | null;
  mandatory_skills: string[];
  preferred_skills: string[];
  responsibilities: string[];
  parsed_requirements_json: Record<string, any>;
  created_at: string;
  updated_at: string;
}

export interface JobCreatePayload {
  title: string;
  company: string;
  raw_description: string;
  url?: string;
  seniority?: string;
  location?: string;
  salary_range?: string;
}

export interface JobMatchResponse {
  job_id: string;
  resume_id: string;
  match_percentage: number;
  matched_skills: string[];
  missing_skills: string[];
  match_rationale: string;
  evidence_citations: string[];
}

File: backend/app/schemas/job.py
"""Pydantic schemas for Job Intelligence."""

import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict, Field


class JobCreate(BaseModel):
    title: str = Field(..., min_length=1, max_length=255)
    company: str = Field(..., min_length=1, max_length=255)
    url: str | None = None
    raw_description: str = Field(..., min_length=10)
    seniority: str | None = "Mid-Senior"
    location: str | None = None
    salary_range: str | None = None


class JobResponse(BaseModel):
    id: uuid.UUID
    candidate_id: uuid.UUID
    title: str
    company: str
    url: str | None
    raw_description: str
    seniority: str
    location: str | None
    salary_range: str | None
    mandatory_skills: list[str]
    preferred_skills: list[str]
    responsibilities: list[str]
    parsed_requirements_json: dict[str, Any]
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class JobMatchResponse(BaseModel):
    job_id: uuid.UUID
    resume_id: uuid.UUID
    match_percentage: int
    matched_skills: list[str]
    missing_skills: list[str]
    match_rationale: str
    evidence_citations: list[str]
